### DISCUSSION TITLE
How do i perform sort with custom script

### DISCUSSION THREAD

**Discussion by ashish64** [Jeroen-G/Explorer#237] [Q&A]
First of all, I want to thank you for creating this amazing package. This is super helpful and easy to use.
I have a quick question i hope someone can point me to right direction, how am i able to pass _script with custom sorting script in my queries?

**Comment by Jeroen-G** [comment]
What would the normal Elastic query look like without PHP etc?

**Reply by ashish64** [reply]
Hey, thank you for getting back.

The elastic query would look something like below. The script itself is not finalised but its gonna be very similar to this. 

```
{
    "query": {
        "bool": {
            "must": [
                {
                    "multi_match": {
                        "query": "fasanvej",
                        "fuzziness": "auto"
                    }
                }
            ],
            "should": [],
            "filter": []
        }
    },
    "sort": {
        "_script": {
            "type": "number",
            "script": {
                "lang": "painless",
                "source":  "def SortOrder = doc['statusOrder'].value;  if (params.sort_order == 'asc') { return SortOrder } else { if ( SortOrder == 0 || SortOrder == 1 ) { return 24 - SortOrder  } else { return  -SortOrder }}",
                "params": {
                    "sort_order": "asc"
                }
            }
        }
    }
}
```

**Reply by Vitsen15** [reply]
I also need such functionality, any updates on this, @Jeroen-G?

**Reply by Jeroen-G** [reply]
You'll have to dig into writing your own sort syntax class for this.

**Reply by Vitsen15** [reply]
Got it, thanks.

**Reply by Vitsen15** [reply]
@Jeroen-G, having troubles with custom syntax usage, it seems like it's impossible to pass custom syntax class to scout query builder `orderBy()` method. Otherwise it will not be composed to be inside root `"sort"` property.

Could you please advice if I can use your library to achieve this or I need to extend scout query builder to be able to pass sorting script  inside the root `"sort"` property?

**Reply by Jeroen-G** [reply]
Sorry, I currently do not have the free time to dive into how this could work for you.

**Reply by Vitsen15** [reply]
Was able to achieve desire search results by using scout query builder `query` method with callback: [Customizing the Eloquent Results Query](https://laravel.com/docs/10.x/scout#customizing-the-eloquent-results-query).

May be somebody finds this useful...
